✨ allow users to customize the attribute used to define the action name

niieani opened the pull request:
## Motivation

`data-test-id` and `data-testid` are the de-facto standard for consistent naming of actionable elements, used for both unit and integration testing. Being forced to duplicate these same names and introduce a vendor-specific attribute (`data-dd-action-name`) is not ideal. 

The reason why a custom attribute is preferred to the default fallback mechanisms is that localized applications will contain strings in different languages, splitting the same action into multiple ones. Additionally, this helps prevent leaking of personal data in case the action is personalized to the content only visible to the user.

<!-- Why are you making this change, what problem does it solve? Include links to relevant tickets. -->

## Changes

Adds the option to customize the attribute name.

<!-- What does this change exactly? Who will be affected? Include relevant screenshots, videos, links. -->

## Testing

Uses the same code paths that already exist, but nonetheless added test cases for these new scenarios.

<!-- How can the reviewer confirm these changes do what you say they do? Are there automated tests? -->

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

bcaudan commented:
Hi @niieani, thanks for submitting this!

Being able to reuse existing attributes to avoid duplication is really interesting, two concerns with the suggested approach:

- current usage of `data-test-id` and `data-testid` could change the data currently collected, so it would be a breaking change
- other customers may use different attributes

So instead of hardcoding these two new attributes, it could be interesting to introduce a new configuration option to provide another attribute name to use.

What do you think?

niieani commented:
Hi @bcaudan, updated and rebased the code based on your review. The attribute is now customizable, but still defaults to the attribute that works today, so it is not a breaking change.

bcaudan commented:
Thanks @niieani!

I took over your PR in #919  to ease the review / merge process.
we'll ping here when it will be available.

bcaudan commented:
Available in 2.16.0
